### ISSUE TITLE
Portability / POSIX compliance

### ISSUE THREAD

**Issue by pkill37** [pkill37/mybb-shetup#2]
I would expect people to run the script on shells other than bash. I myself use zsh, and it's certainly quite popular in the web development community. As such I think it's important to make the script as portable as possible.

I just gave it a run on zsh and it's totally failing. So we need to fix it and make it work on the various shells out there. I did some research and I've already committed some stuff. I'll just dump the links here if it's any useful. They're all about bash technically, but you'll often find POSIX-compliant solutions to stuff.

http://andreinc.net/2011/09/04/bash-scripting-best-practice/
http://mywiki.wooledge.org/BashPitfalls
http://mywiki.wooledge.org/BashFAQ

**Comment by JoshHarmon** [comment]
Yeah --- I don't use zsh (or any other shell)... so I'm pretty much out on this one, I think.  

## 

Josh Harmon
Sent with Sparrow (http://www.sparrowmailapp.com/?sig)

**Comment by pkill37** [comment]
Made some significant progress on this in the last couple commits.

**Comment by JoshHarmon** [comment]
How far along would you say this is? I myself have never ventured into POSIX compliance standards, so I wouldn't know.

**Comment by pkill37** [comment]
I kinda had to learn myself. Right now, the `pause()` function doesn't work under zsh. `read -p` is a bash thing, zsh requires a different syntax (hence why I made the generic prompt functions to abstract that away). I'm looking into it. In the meantime, feel free to fix it if you have any idea how.

I haven't tested any further, but I did considerable research and I think the code is fine. I'm not aiming for full portability in the sense of POSIX compliance, bash and zsh compatibility is cool with me. Very few people use anything else. Nonetheless the script should work under the older Bourne shell and stuff.
